### PR TITLE
🐳Change the pycache folder path to avoid re-create container

### PR DESCRIPTION
**Pull request by trajepl**
## Describe your changes
Please refer to this discussion:
https://github.com/microsoft/Olive/pull/384#discussion_r1244710954 

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: olive/systems/docker/docker_system.py [hunk 1]
```diff
@@ -149,43 +150,32 @@ def _run_container(self, tempdir, model: OliveModel, metrics: List[Metric], cont
...
+            command=eval_command,
+            volumes=volumes_list,
+            detach=True,
+            environment=environment,
+            **run_command,
+        )
+        for line in container.logs(stream=True):
+            logger.info(line.strip().decode())
+        exit_code = container.wait()["StatusCode"]
+        container.remove()
```
Comment (trajepl) [inline olive/systems/docker/docker_system.py:173]: Please help double check if we can remove the container once the evaluate done.

I comment this out in last PR as I was thinking if we should keep the container for debug. But just found too many containers were created and it is hard to start as the mount path in host may be changed.
image

Comment (jambayk) [reply olive/systems/docker/docker_system.py:173]: I think removing the container is a good idea. I found that I had a lot of stopped containers on my machine when trying to remove images.

File: olive/systems/docker/docker_system.py [hunk 2]
```diff
@@ -149,43 +150,32 @@ def _run_container(self, tempdir, model: OliveModel, metrics: List[Metric], cont
 
         run_command = docker_utils.create_run_command(run_params=self.run_params)
 
-        try:
-            logger.debug(f"Running container with eval command: {eval_command}")
-            logger.debug(f"The volumes list is {volumes_list}")
-            container = self.docker_client.containers.run(
-                image=self.image, command=eval_command, volumes=volumes_list, detach=True, **run_command
+        environment = run_command.pop("environment", {})
+        envs_dict = {"PYTHONPYCACHEPREFIX": "/tmp"}
```
[the remark this answers is not in the document]
Comment (jambayk) [reply olive/systems/docker/docker_system.py:154]: If so, maybe you can also remove the `dev_mount_cleanup.py` script in this PR.

[the remark this answers is not in the document]
Comment (guotuofeng) [reply olive/systems/docker/docker_system.py:154]: yes, it seems the dev_mount_cleanup.py could be removed

**Review by guotuofeng — APPROVED** [review]

**Pull request merged**